src/components/common/FilterGroup/filters/FilterMultipleSelection.tsx

```diff
@@ -33,16 +32,33 @@ export default function MultipleSelection(props: Props): JSX.Element {
     setSelections(props.values);
   }, [props.values]);
 
-  const handleChange = (value: string | null) => {
-    const updatesValues = [...selections];
+  const optionsMap = useMemo(
+    () =>
+      new Map<string, string>(
+        props.options.filter((opt) => opt.value && opt.label).map((opt) => [opt.value!, opt.label!])
+      ),
+    [props.options]
+  );
 
+  const onAdd = (value: string | null) => {
+    const updatesValues = [...selections];
     const valueIndex = updatesValues.findIndex((v) => v === value);
     if (valueIndex < 0) {
       updatesValues.push(value);
-    } else {
+    }
+    updateFilters(updatesValues);
```

Comment:
move this into the if block on line 46?

Reply:
Good call; no need to perform this op if the updatesValues is unchanged 👍